keanu-python/tests/test_stats.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pytest
+
+from itertools import islice
+from keanu import Model, stats, KeanuRandom
+from keanu.algorithm import sample, generate_samples
+from keanu.vertex import Uniform
+import pandas as pd
+
+
+def test_can_get_correct_autocorrelation() -> None:
+    x = [1, 2, 3, 4, 5, 6, 7, 8]
+    x_list = [np.array(a, float) for a in x]
+    autocorr = stats.autocorrelation(x_list)
+    expected = [1., 0.625, 0.27380952, -0.0297619, -0.26190476, -0.39880952, -0.41666667, -0.29166667]
+    np.testing.assert_almost_equal(autocorr, expected)
+
+
+def test_can_get_autocorrelation_for_samples() -> None:
+    with Model() as m:
+        m.uniform = Uniform(0, 1000)
+    net = m.to_bayes_net()
+    samples = sample(net=net, sample_from=net.get_latent_vertices(), algo="metropolis", draws=1000)
+    valid_key = list(samples.keys())[0]
+    autocorr = stats.autocorrelation(samples.get(valid_key))  # type: ignore
```

Comment:
nit: might be a bit silly, it'd be nice to add some assertions on the return type and value (e.g. autocorrelation of lag 0 is greater than or equal to the rest).

Reply:
I agree with the return type. I will assert that it is an `ndarray`. I am hesitant to add anymore value checks on the python side as we are already doing so on the Java side. It could unnecessarily make tests longer to run. Happy to be convinced otherwise though.

Reply:
Added a check on the datatype. Check that it is an ndarray and that it is the same type as the data it is given.